src/Tokens/Models/UserIdToken/UserIdToken.php

```
@@ -162,6 +175,11 @@ private function parseToken(): array
             throw new InvalidUserIdTokenException(sprintf('Unable to parse JSON string %s.', $payloadJsonString));
         }
 
-        return (array)$payload;
+        return is_array($payload) ? $payload : (array)$payload;
```

The recommendation to check if it's an array was more to enforce that the payload was parsed properly. If it's not an array already, something went wrong and we should throw an exception. The `$payloadJsonString` could be something like `"true"`, which parses to a (bool) `true` value, and then when cast to an array turns into `[true]`.